core/src/test/scala/integration/kafka/api/LogDirFailureTest.scala

```
@@ -60,6 +61,37 @@ class LogDirFailureTest extends IntegrationTestHarness {
   }
 
   @Test
+  def brokerWithOldIBPShouldHaltOnLogDirFailure() {
+    @volatile var statusCodeOption: Option[Int] = None
+    Exit.setHaltProcedure { (statusCode, _) =>
+      statusCodeOption = Some(statusCode)
+      throw new IllegalArgumentException
```

Do we need to throw an exception here? It seems that we could just check statusCodeOption after the server is started.

---

We need to throw exception here because the signature of `setHaltProcedure` is `setHaltProcedure(haltProcedure: (Int, Option[String]) => Nothing): Unit`. The return value of the function is `Nothing`, not Unit.